src/main.rs

```diff
@@ -54,13 +75,53 @@ fn main() {
     let mut mouse_mode = MouseMode::Cursor;
     let mut modifiers = winit::event::ModifiersState::empty();
 
+    let mut ui_state = UiState::default();
+    let mut prev_frame_time = Instant::now();
+    let mut last_cursor = None;
+
+
+    let mut hidpi_factor = 1.0;
+    let mut imgui_context = imgui::Context::create();
+    let mut platform = imgui_winit_support::WinitPlatform::init(&mut imgui_context);
+    platform.attach_window(
+        imgui_context.io_mut(),
+        &window,
+        imgui_winit_support::HiDpiMode::Default,
+    );
+    imgui_context.set_ini_filename(None);
+
+    let font_size = (13.0 * hidpi_factor) as f32;
+    imgui_context.io_mut().font_global_scale = (1.0 / hidpi_factor) as f32;
+
+    imgui_context.fonts().add_font(&[FontSource::DefaultFontData {
+        config: Some(imgui::FontConfig {
+            oversample_h: 1,
+            pixel_snap_h: true,
+            size_pixels: font_size,
+            ..Default::default()
+        }),
+    }]);
+
+    let mut imgui_renderer = imgui_wgpu::Renderer::new(
+        &mut imgui_context,
+        &renderer.device,
+        &mut renderer.queue,
+        renderer.sc_desc.format,
+        None,
+    );
```

Comment:
All this should be in it's own function call for setting up the im gui window stuff.